caffe2/core/blob_serialization.h

```diff
@@ -23,6 +23,38 @@ constexpr auto kTensorBlobType = "Tensor";
 // String used to separate chunk id from the blob name when storing in DB
 constexpr auto kChunkIdSeparator = "#%";
 
+/**
+ * Serializes the given blob, if possible. Note that this serialization uses
+ * the registration mechanism and one has to implement specific serialization
+ * approaches for specific classes. Acceptor should take care of writing data
+ * to the actual storage.
+ */
+void SerializeBlob(
+    const Blob& blob,
+    const string& name,
+    BlobSerializerBase::SerializationAcceptor acceptor,
+    int chunk_size = kDefaultChunkSize);
+
+/**
+ * @brief Convenience function to serialize a blob to a string.
+ *
+ * This is a conveinence function to serialize small Blobs that produce
+ * manageable serialized strings. To serialize big blobs such as
+ * large sparse tensors, use the fully-functional interface in
+ * blob_serializer_base.h.
+ *
+ * NOTE: this function doesn't do chunking and might break with big tensors.
+ */
+string SerializeBlob(const Blob& blob, const string& name);
+
+/**
+ * Deserializes from a string containing either BlobProto or TensorProto. If
+ * the deserialization fails, the content in the blob should no longer be
+ * trusted.
```

Comment:
This a really weird way to setup the API now that you have an out-of-line function. Why not just return a blob? Blobs are pointery things, so returning a new one ought to be fine, right?

Reply:
(I do understand that this would lead to more refactoring)

Reply:
My first approach was to return a Blob but there were some call sites that would have been hard to change. I'm going with this for now because it achieves what I need. We can consider further refactoring this later, or just replace it with IValue entirely.